consensus-types/hdiff/state_diff.go

```diff
@@ -0,0 +1,2101 @@
+	previousJustifiedCheckpoint *ethpb.Checkpoint                                           // override.
+	currentJustifiedCheckpoint  *ethpb.Checkpoint                                           // override.
+	finalizedCheckpoint         *ethpb.Checkpoint                                           // override.
+	// Altair Fields
+	previousEpochParticipation []byte               // override.
+	currentEpochParticipation  []byte               // override.
+	inactivityScores           []uint64             // override.
+	currentSyncCommittee       *ethpb.SyncCommittee // override.
+	nextSyncCommittee          *ethpb.SyncCommittee // override.
+	// Bellatrix
+	executionPayloadHeader interfaces.ExecutionData // override.
+	// Capella
+	nextWithdrawalIndex          uint64                     // override.
+	nextWithdrawalValidatorIndex primitives.ValidatorIndex  // override.
+	historicalSummaries          []*ethpb.HistoricalSummary // append only.
+	// Electra
+	depositRequestsStartIndex     uint64           // override.
+	depositBalanceToConsume       primitives.Gwei  // override.
+	exitBalanceToConsume          primitives.Gwei  // override.
+	earliestExitEpoch             primitives.Epoch // override.
+	consolidationBalanceToConsume primitives.Gwei  // override.
+	earliestConsolidationEpoch    primitives.Epoch // override.
+
+	pendingDepositIndex            uint64                            // override.
+	pendingPartialWithdrawalsIndex uint64                            // override.
+	pendingConsolidationsIndex     uint64                            // override.
+	pendingDepositDiff             []*ethpb.PendingDeposit           // override.
+	pendingPartialWithdrawalsDiff  []*ethpb.PendingPartialWithdrawal // override.
+	pendingConsolidationsDiffs     []*ethpb.PendingConsolidation     // override.
+	// Fulu
+	proposerLookahead []uint64 // override
+}
+
+type hdiff struct {
+	stateDiff      *stateDiff
+	validatorDiffs []validatorDiff
+	balancesDiff   []int64
+}
+
+// validatorDiff is a type that represents a difference between two validators.
+type validatorDiff struct {
+	Slashed                    bool             // new value (here because of alignement)
+	index                      uint32           // override.
+	PublicKey                  []byte           // override.
+	WithdrawalCredentials      []byte           // override.
+	EffectiveBalance           uint64           // override.
+	ActivationEligibilityEpoch primitives.Epoch // override
+	ActivationEpoch            primitives.Epoch // override
+	ExitEpoch                  primitives.Epoch // override
+	WithdrawableEpoch          primitives.Epoch // override
+}
+
+var (
+	errDataSmall = errors.New("data is too small")
+)
+
+const (
+	nilMarker                      = byte(0)
+	notNilMarker                   = byte(1)
+	forkLength                     = 2*fieldparams.VersionLength + 8
+	blockHeaderLength              = 8 + 8 + 3*fieldparams.RootLength
```

Comment:
I would love some comments around these numbers or a link to spec

Reply:
Fixed in [228b2ea](https://github.com/OffchainLabs/prysm/pull/15250/commits/228b2ea1d6c8bf06b766b16513f9970eac1a3d2c)